gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -616,33 +457,27 @@ func (dir *gcsDir) Render(out http.ResponseWriter, inPath string) {
 
 // Record represents a single "Contents" entry in a GCS bucket.
 type Record struct {
-	Name  string `xml:"Key"`
-	MTime string `xml:"LastModified"`
-	Size  int64  `xml:"Size"`
+	Name  string
+	MTime time.Time
+	Size  int64
 	isDir bool
 }
 
 // Render writes HTML representing this Record to the provided output.
 func (rec *Record) Render(out http.ResponseWriter, inPath string) {
-	mtime := "<unknown>"
-	ts, err := time.Parse(time.RFC3339, rec.MTime)
-	if err == nil {
-		mtime = ts.Format("02 Jan 2006 15:04:05")
-	}
-	var url, size string
-	if rec.isDir {
-		url = gcsPath + inPath + rec.Name
-		size = "-"
-	} else {
-		url = gcsBaseURL + inPath + rec.Name
-		size = fmt.Sprintf("%v", rec.Size)
```

Comment:
Why/how is this logic ok to omit? It looks like the `isDir` field isn't used at all anymore and this always constructs the URL assuming the record is a directory. (but formats the size as if its a file?)

Reply:
Before, the URL of the files was pointing directly to the google link. That was feasible because all the links were public. However now we need to get the contents of the files using a potential authenticated storage client.  So we omit the current logic and we can make the files point to the same gcsweb instance, and with its turn, it will handle the object as a file and output its contents. The storage client can understand if the requested object is a file or not by getting the attributes [here](https://github.com/kubernetes/test-infra/blob/a5a26e60f03f147274cf9fa2056c687244e8b29e/gcsweb/cmd/gcsweb/gcsweb.go#L311)